src/plugins/workflow_designer/src/library/DocWorkers.cpp

```diff
@@ -873,6 +875,134 @@ bool MSAWriter::isStreamingSupport() const {
     return false;
 }
 
+/*************************************
+ * UgeneDBWriter
+ *************************************/
+void UgeneDBWriter::data2doc(Document *doc, const QVariantMap &data) {
+    data2document(doc, data, context);
+}
+
+void UgeneDBWriter::storeEntry(IOAdapter *io, const QVariantMap &data, int entryNum) {
+    streamingStoreEntry(format, io, data, context, entryNum);
+}
+
+void UgeneDBWriter::data2document(Document *doc, const QVariantMap &data, WorkflowContext *context) {
+    QScopedPointer<U2SequenceObject> seqObj(nullptr);
+    U2SequenceObject *dna = nullptr;
+    QString annotationName;
+
+    if (data.contains(BaseSlots::DNA_SEQUENCE_SLOT().getId())) {
+        SharedDbiDataHandler seqId = data[BaseSlots::DNA_SEQUENCE_SLOT().getId()].value<SharedDbiDataHandler>();
+        seqObj.reset(StorageUtils::getSequenceObject(context->getDataStorage(), seqId));
+        SAFE_POINT(nullptr != seqObj.data(), tr("UgeneDB writer: NULL sequence object"), );
```

Comment:
1) do not use toda style.
2) do not use translation in SAFE_POINTS. If this is a user-facing log use CHECK